packages/firestore/test/unit/specs/limbo_spec.test.ts

```diff
@@ -927,6 +928,48 @@ describeSpec('Limbo Documents:', [], () => {
     }
   );
 
+  // Reproduce the bug in b/238823695
+  specTest(
+    'Limbo resolution should wait for full re-query result if there is an existence filter mismatch ',
+    [],
+    () => {
+      const query1 = query('collection');
+      const docA = doc('collection/a', 1000, { v: 1 });
+      const docB = doc('collection/b', 1000, { v: 2 });
+      return (
+        spec()
+          .userListens(query1)
+          .watchAcksFull(query1, 1000, docA, docB)
+          .expectEvents(query1, { added: [docA, docB] })
+          // Simulate that the client loses network connection.
+          .disableNetwork()
+          .expectEvents(query1, { fromCache: true })
+          .enableNetwork()
+          .restoreListen(query1, 'resume-token-1000')
+          .watchAcks(query1)
+          // DocB is deleted in the next sync.
+          .watchFilters([query1], [docA.key])
+          // Bugged behavior: Missing watchCurrent here will move
```

Comment:
nit: I think you can remove this comment about "Bugged behavior" since the problematic behavior does not occur anymore.